src/views/AddItem.jsx

```diff
@@ -1,3 +1,115 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
+export function AddItem({ listToken }) {
+	const soon = 7;
+	const kindaSoon = 14;
+	const notSoon = 30;
+
+	const [itemName, setItemName] = useState('');
+	const [nextPurchase, setNextPurchase] = useState(0);
+	const [submissionConfirmation, setsubmissionConfirmation] = useState('');
+
+	const handleChange = (e) => {
+		setItemName(e.target.value);
+	};
+
+	const submitForm = (e) => {
+		e.preventDefault();
+
+		//form validation
+		if (itemName === '') {
+			alert('Please specify the name of the item');
+			return;
+		}
+
+		if (nextPurchase === 0) {
+			alert(
+				"Please select how soon you'll be purchasing the item again (soon/Kinda Soon/Not Soon",
+			);
+			return;
+		}
+
+		let itemData = {
+			itemName: itemName,
+			daysUntilNextPurchase: nextPurchase,
+		};
+
+		addItem(listToken, itemData)
+			.then(() => {
+				//clear form
+				setItemName('');
+				setNextPurchase(0);
+
+				// put a note that the form was submitted (and erase after 5 seconds)
+				setsubmissionConfirmation(
```

Comment:
```suggestion
				setSubmissionConfirmation(
```

Reply:
Although, I realise if you did this, you'd have to rely on the refactor tools in VSCode, or just a careful search and replace to keep the capitlization correct. Your call, just thought I'd point it out :)